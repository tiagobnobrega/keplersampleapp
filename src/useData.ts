import {useQuery} from '@tanstack/react-query';

function delay(t: number, cb: () => void) {
  return new Promise((resolve) =>
    setTimeout(() => {
      cb();
      resolve(undefined);
    }, t),
  );
}
export const useData = () => {
  const {data: firstData, isLoading} = useQuery({
    queryKey: ['1'],
    queryFn: async () => {
      await delay(5000, () => {});
      return [{id: 'A'}, {id: 'B'}];
    },
  });

  const {data: secondData, isLoading: isLoadingSecond} = useQuery({
    queryKey: ['2'],
    queryFn: async () => {
      await delay(5500, () => {});
      return firstData?.map((fd, index) => {
        return `Second ${fd.id}-${index + 1}`;
      });
    },
    enabled: !!firstData,
  });

  return {firstData, secondData, isLoading: isLoading || isLoadingSecond};
};
