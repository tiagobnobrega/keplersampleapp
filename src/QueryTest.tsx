import {StyleSheet, Text, View} from 'react-native';
import React from 'react';
import {useQuery} from '@tanstack/react-query';
import {useData} from './useData';

export const QueryTest = () => {
  const styles = getStyles();
  const {firstData, secondData, isLoading} = useData();

  return (
    <View style={styles.container}>
      <Text style={styles.text}>Loading: {isLoading ? 'loading' : 'done'}</Text>
      <View>
        {firstData?.map((d) => (
          <View key={d.id}>
            <Text style={styles.text}>first data {d.id}</Text>
          </View>
        ))}
      </View>
      <View>
        {secondData?.map((d) => (
          <View key={d}>
            <Text style={styles.text}>{d}</Text>
          </View>
        ))}
      </View>
    </View>
  );
};

const getStyles = () =>
  StyleSheet.create({
    btn: {
      width: 200,
    },
    background: {
      color: 'white',
      flex: 1,
      flexDirection: 'column',
    },
    container: {
      flex: 6,
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: 'red',
    },
    headerContainer: {
      marginLeft: 200,
    },
    headerText: {
      color: 'white',
      fontSize: 80,
      marginBottom: 10,
    },
    subHeaderText: {
      color: 'white',
      fontSize: 40,
    },
    links: {
      flex: 1,
      flexDirection: 'column',
      justifyContent: 'space-around',
      height: 600,
    },
    image: {
      flex: 1,
      paddingLeft: 150,
    },
    textContainer: {
      justifyContent: 'center',
      flex: 1,
      marginLeft: 190,
    },
    text: {
      color: 'white',
      fontSize: 40,
    },
  });
