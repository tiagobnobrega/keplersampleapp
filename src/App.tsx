/*
 * Copyright (c) 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */

import React, {useCallback, useState, useSyncExternalStore} from 'react';
import {
  ImageBackground,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
import {QueryTest} from './QueryTest';

function delay(t: number, cb: () => void) {
  return new Promise((resolve) =>
    setTimeout(() => {
      cb();
      resolve(undefined);
    }, t),
  );
}

export const createTodosStore = (name = 'Todo') => {
  let nextId = 0;
  let todos = [{id: nextId++, text: `${name} #1`}];
  let listeners: any[] = [];

  function emitChange() {
    for (const listener of listeners) {
      listener();
    }
  }
  return {
    async addTodo(t = 2000) {
      await delay(t, () => {
        todos = [...todos, {id: nextId++, text: `${name} #` + nextId}];
        emitChange();
      });
    },
    subscribe(listener: any) {
      listeners = [...listeners, listener];
      return () => {
        listeners = listeners.filter((l) => l !== listener);
      };
    },
    getSnapshot() {
      return todos;
    },
  };
};

const myTodosStore = createTodosStore();
const otherTodosStore = createTodosStore('Other');

export const App = () => {
  const queryClient = new QueryClient();
  const [syncState, setSyncState] = useState('');
  const styles = getStyles();

  const myTodos = useSyncExternalStore(
    myTodosStore.subscribe,
    myTodosStore.getSnapshot,
  );

  const otherTodos = useSyncExternalStore(
    otherTodosStore.subscribe,
    otherTodosStore.getSnapshot,
  );

  const handlePress = useCallback(() => {
    myTodosStore.addTodo();
    otherTodosStore.addTodo(1000);
    setSyncState(Date.now().toString());
  }, []);
  return (
    <ImageBackground
      source={require('./assets/background.png')}
      style={styles.background}>
      <View style={styles.container}>
        <TouchableOpacity style={styles.btn} onPress={handlePress}>
          <Text style={styles.text}>Add todo</Text>
        </TouchableOpacity>

        <View>
          {myTodos.map((todo) => (
            <Text style={styles.text} key={todo.id}>
              {todo.text}
            </Text>
          ))}
        </View>

        <View>
          {otherTodos.map((todo) => (
            <Text style={styles.text} key={todo.id}>
              {todo.text}
            </Text>
          ))}
        </View>
        <View>
          <Text style={styles.text}>{syncState}</Text>
        </View>
      </View>
      <QueryClientProvider client={queryClient}>
        <QueryTest />
      </QueryClientProvider>
    </ImageBackground>
  );
};

const getStyles = () =>
  StyleSheet.create({
    btn: {
      width: 200,
    },
    background: {
      color: 'white',
      flex: 1,
      flexDirection: 'column',
    },
    container: {
      flex: 6,
      flexDirection: 'row',
      alignItems: 'center',
    },
    headerContainer: {
      marginLeft: 200,
    },
    headerText: {
      color: 'white',
      fontSize: 80,
      marginBottom: 10,
    },
    subHeaderText: {
      color: 'white',
      fontSize: 40,
    },
    links: {
      flex: 1,
      flexDirection: 'column',
      justifyContent: 'space-around',
      height: 600,
    },
    image: {
      flex: 1,
      paddingLeft: 150,
    },
    textContainer: {
      justifyContent: 'center',
      flex: 1,
      marginLeft: 190,
    },
    text: {
      color: 'white',
      fontSize: 40,
    },
  });
