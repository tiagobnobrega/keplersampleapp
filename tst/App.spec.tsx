/**
 * Copyright (c) 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */

import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import * as React from 'react';

import {App} from '../src/App';

describe('Template App Snapshot tests', () => {
  it('Initial App screen', () => {
    const screen = render(<App />);
    expect(screen).toMatchSnapshot();
  });

  it('App screen after link press', () => {
    const screen = render(<App />);
    const button = screen.getByTestId('sampleLink');
    fireEvent.press(button);
    expect(screen).toMatchSnapshot();
  });
});
