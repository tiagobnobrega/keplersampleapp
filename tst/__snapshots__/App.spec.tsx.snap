// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template App Snapshot tests Initial App screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "color": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "height": 600,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        style={
          Object {
            "marginLeft": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 80,
              "marginBottom": 10,
            }
          }
        >
          Hello World!
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 40,
            }
          }
        >
          Select one of the options below to start your Kepler journey ðŸš€
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "transparent",
            "borderBottomWidth": 5,
            "opacity": 1,
            "paddingBottom": 10,
            "width": 420,
          }
        }
        testID="sampleLink"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 200,
              "width": 300,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/star.png",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 45,
                "marginLeft": 30,
              }
            }
          >
            Learn
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "transparent",
            "borderBottomWidth": 5,
            "opacity": 1,
            "paddingBottom": 10,
            "width": 420,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 200,
              "width": 300,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/star.png",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 45,
                "marginLeft": 30,
              }
            }
          >
            Build
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "transparent",
            "borderBottomWidth": 5,
            "opacity": 1,
            "paddingBottom": 10,
            "width": 420,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 200,
              "width": 300,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/star.png",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 45,
                "marginLeft": 30,
              }
            }
          >
            Support
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 150,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/kepler.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 190,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 40,
        }
      }
    >
      ðŸ’¡ Edit App.tsx to change this screen and then come back to see your edits
    </Text>
  </View>
</View>
`;

exports[`Template App Snapshot tests App screen after link press 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "color": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "height": 600,
          "justifyContent": "space-around",
        }
      }
    >
      <View
        style={
          Object {
            "marginLeft": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 80,
              "marginBottom": 10,
            }
          }
        >
          Hello World!
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 40,
            }
          }
        >
          Select one of the options below to start your Kepler journey ðŸš€
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "transparent",
            "borderBottomWidth": 5,
            "opacity": 1,
            "paddingBottom": 10,
            "width": 420,
          }
        }
        testID="sampleLink"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 200,
              "width": 300,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/star.png",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 45,
                "marginLeft": 30,
              }
            }
          >
            Learn
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "transparent",
            "borderBottomWidth": 5,
            "opacity": 1,
            "paddingBottom": 10,
            "width": 420,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 200,
              "width": 300,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/star.png",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 45,
                "marginLeft": 30,
              }
            }
          >
            Build
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderBottomColor": "transparent",
            "borderBottomWidth": 5,
            "opacity": 1,
            "paddingBottom": 10,
            "width": 420,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 200,
              "width": 300,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/star.png",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 45,
                "marginLeft": 30,
              }
            }
          >
            Support
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 150,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/learn.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 190,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 40,
        }
      }
    >
      ðŸ’¡ Edit App.tsx to change this screen and then come back to see your edits
    </Text>
  </View>
</View>
`;
